internal/cli/atlas/search/nodes/spec_file.go

```diff
@@ -0,0 +1,35 @@
+// Copyright 2024 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// TODO: remove file when merging to master
+
+package nodes
+
+import (
+	"fmt"
+
+	"github.com/mongodb/mongodb-atlas-cli/internal/file"
+	"github.com/spf13/afero"
+	atlasv2 "go.mongodb.org/atlas-sdk/v20231115007/admin"
+)
+
+// Load []atlasv2.ApiSearchDeploymentSpec from a given file.
+func loadAPISearchDeploymentSpec(fs afero.Fs, filename string) (*[]atlasv2.ApiSearchDeploymentSpec, error) {
+	spec := new([]atlasv2.ApiSearchDeploymentSpec)
+	if err := file.Load(fs, filename, spec); err != nil {
+		return nil, fmt.Errorf("failed to parse JSON file due to: %w", err)
+	}
```

Comment:
I'm not understanding this function is the spected file an array of spec?
```json
[
{

    "instanceSize": "S20_HIGHCPU_NVME",
    "nodeCount": 2

}]
```
https://www.mongodb.com/docs/atlas/reference/api-resources-spec/v2/#tag/Atlas-Search/operation/updateAtlasSearchDeployment
the api documents this endpoint as 
```
{

    "specs": 

[

        {
            "instanceSize": "S20_HIGHCPU_NVME",
            "nodeCount": 2
        }
    ]

}
```

Consider what would happen if I do 
```zsh
(TMPSUFFIX=.json; atlas clusters search node create newName -f =(atlas clusters search node describe oldName -o json))
```

Reply:
> I'm not understanding this function is the spected file an array of spec?
> 
> ```json
> [
> {
> 
>     "instanceSize": "S20_HIGHCPU_NVME",
>     "nodeCount": 2
> 
> }
> ]
> ```
> 
> https://www.mongodb.com/docs/atlas/reference/api-resources-spec/v2/#tag/Atlas-Search/operation/updateAtlasSearchDeployment the api documents this endpoint as
> 
> ```json
> {
>     "specs": [
>         {
>             "instanceSize": "S20_HIGHCPU_NVME",
>             "nodeCount": 2
>         }
>     ]
> }
> ```

That's correct, the file is an array of spec. The example you described looks correct.

> Consider what would happen if I do
> 
> ```shell
> (TMPSUFFIX=.json; atlas clusters search node create newName -f =(atlas clusters search node describe oldName -o json))
> ```

This command won't work the way things are currently implemented/described in the spec.

Reply:
>This command won't work the way things are currently implemented/described in the spec.

why not? we should make it work

Reply:
I can make it work, but I'll do it in the create PR (#2643), since I planned on deleting the file in this branch.

Reply:
Fixed here: https://github.com/mongodb/mongodb-atlas-cli/pull/2643/commits/36e803e545239715eb9e23c5eb3f52a657920525

Reply:
Should be resolved in this branch too now